File: HackerRank/Algorithms/2. Implementation/Minimum_distances.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var n = parseInt(readLine());
    A = readLine().split(' ');
    A = A.map(Number);

    let i, j, minimum = 10000;
    for (i = 0; i < n; i++) {
      for (j = i + 1; j < n; j++) {
        if (A[i] == A[j]) {
          minimum = Math.min(minimum, Math.abs(i - j));
        }
      }
    }

    if (minimum == 10000) {
      console.log(-1);
    } else {
      console.log(minimum);
    }
}

File: HackerRank/Algorithms/2. Implementation/Minimum_distances.py
#!/bin/python3

n = int(input().strip())
A = [int(A_temp) for A_temp in input().strip().split(' ')]

minimum = 10000;
for i in range(0, n):
    for j in range(i + 1, n):
        if (A[i] == A[j]):
            minimum = min(minimum, abs(i - j))

if minimum != 10000:
    print (minimum)
else:
    print (-1)
